archivist/assets.py

```diff
@@ -250,9 +251,10 @@ def create_if_not_exists(
 
         # any attachments ?
         if attachments is not None:
-            data["attributes"]["arc_attachments"] = [
-                self._archivist.attachments.create(a) for a in attachments
-            ]
+            for a in attachments:
+                data["attributes"][
+                    a["attachment"]
```

Comment:
How do we know that the key `attachment` will be present? In the YAML example further up in the function, it suggests that filename will be there - but I see no mention of this key.